### PR TITLE
feat:  add auto-generation of state proto

### PR DESCRIPTION
**Pull request by abeatrix**
### Related Issue

**Issue:** #XXXX

### Description

Add lint-staged hook to automatically regenerate proto/cline/state.proto when src/shared/storage/state-keys.ts changes. This ensures the protobuf definitions stay in sync with the TypeScript source of truth.

Changes:
- Add generate-state-proto.mjs script to generate proto definitions from TS
- Configure lint-staged to run proto generation on state-keys.ts changes
- Update state.proto with regenerated field numbers and new OpenTelemetry fields

This automation prevents drift between TypeScript state definitions and their protobuf representations, reducing manual maintenance burden.

### Type of Change

-   [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
-   [ ] ✨ New feature (non-breaking change which adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] ♻️ Refactor Changes
-   [ ] 💅 Cosmetic Changes
-   [ ] 📚 Documentation update
-   [x] 🏃 Workflow Changes

### Pre-flight Checklist

-   [x] Changes are limited to a single feature, bugfix or chore (split larger changes into separate PRs)
-   [x] Tests are passing (`npm test`) and code is formatted and linted (`npm run format && npm run lint`)
-   [ ] I have created a changeset using `npm run changeset` (required for user-facing changes)
-   [x] I have reviewed [contributor guidelines](https://github.com/cline/cline/blob/main/CONTRIBUTING.md)

### Additional Notes

----

> [!IMPORTANT]
> Automates `state.proto` generation from TypeScript definitions with a new script and lint-staged hook, updating field numbers and adding OpenTelemetry fields.
> 
>   - **Automation**:
>     - Adds `generate-state-proto.mjs` script to auto-generate `proto/cline/state.proto` from `src/shared/storage/state-keys.ts`.
>     - Configures `lint-staged` to run the script on changes to `state-keys.ts`.
>   - **Proto Updates**:
>     - Updates `state.proto` with new field numbers and OpenTelemetry fields.
>   - **Misc**:
>     - Ensures `state.proto` remains in sync with TypeScript definitions, reducing manual updates.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=cline%2Fcline&utm_source=github&utm_medium=referral)<sup> for 289824ba2b449bd36579a4b475ac1702b07bd07e. You can [customize](https://app.ellipsis.dev/cline/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: proto/cline/state.proto [hunk 1]
```diff
@@ -54,184 +54,229 @@ message AutoApprovalSettings {
...
-  optional string groq_api_key = 32;
-  optional string huawei_cloud_maas_api_key = 33;
-  optional string baseten_api_key = 34;
-  optional string vercel_ai_gateway_api_key = 35;
-  optional string dify_api_key = 36;
-  optional string oca_api_key = 37;
-  optional string oca_refresh_token = 38;
-  optional string hicap_api_key = 39;
-  optional string mcp_oauth_secrets = 40;
+  optional string cline_account_id = 2;
```
Comment (maxpaulus43) [inline proto/cline/state.proto]: please check these are all exactly the same as the ones removes

**Comment by abeatrix** [comment]
> Thanks @abeatrix! I appreciate the automation here — my understanding is this adds a script + lint-staged hook to regenerate `proto/cline/state.proto` from `state-keys.ts` to prevent drift.
> 
> That said, I’m worried about protobuf compatibility in the generated output:
> 
> * The regenerated proto renumbers many existing fields (e.g., `open_router_api_key` and others). Field numbers are part of the wire format, so changing them can break persisted state and interop.
> * The script also renames `mcpOAuthSecrets` to `mcp_o_auth_secrets` (extra underscore), which doesn’t match the existing field name and could break serialization/deserialization.
> 
> Can we preserve existing field names/numbers (or add explicit mapping overrides for OAuth and any other mismatches) and confirm this won’t break existing data? Also, the Test Procedure section is empty — a quick note on how you validated the generator would help.
> 
> Happy to re-review once those are addressed. Thanks!

Thanks for the review @arafatkatze ! The proto definitions here are used for the gRPC communication layer between the extension and webview, not for disk persistence, and state is stored as JSON strings, not binary protobuf, so there's no external service expecting these specific field numbers.

I think this PR actually address the second issue you mentioned with the old `mcp_oauth_secrets` which was not correctly formatted.  `mcpOAuthSecrets` is translated to `mcp_o_auth_secrets` in proto buf where `mcp_oauth_secrets` => `mcpOauthSecrets` which does not match how it is stored in our state-keys.

The auto-generated ones are adding the state-keys that are currently missing
